packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -292,55 +310,68 @@ async function doEmit(mainTspFile: string, emitter: Emitter) {
     const content = await readFile(tspConfigFile);
     configYaml = parseDocument(content.toString());
```

Comment:
parseDocument may throw exception. Please create a tryParseYaml util method in util.ts and use it in the file and handle the negative case properly